fix(apigateway): SpecRestApi ignores disableExecuteApiEndpoint property

caldermc-at-amazon opened the pull request:
Fixes [#21295](https://github.com/aws/aws-cdk/issues/21295)

I came upon this bug when trying to create a `SpecRestApi` with its execute API endpoint disabled so that it could only be reached via a custom domain name. I set the `disableExecuteApiEndpoint` property to `true` (because it defaults to `false`) but the endpoint was still enabled. Upon diving deeper I realized this was due to an inconsistency: both the `RestApi` and `SpecRestApi` constructors create a `CfnRestApi`, but only the `RestApi` constructor passes along the property.

When looking at the commit that introduced the `disableExecuteApiEndpoint` property into the `RestApiBaseProps` (used by both `RestApi` and `SpecRestApi`) it seems like failing to pass it along to the `CfnRestApi` in the `SpecRestApi` constructor was simply an oversight. This commit fixes the issue by passing `disableExecuteApiEndpoint` into the `CfnRestApi` in the `SpecRestApi` constructor. I am not aware of any alternatives that I could have considered for this fix.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

TheRealAmazonKendra commented:
Please make sure that your PR body describes the problem the PR is solving, and the design approach and alternatives considered. Explain why the PR solves the problem. A link to an issue is helpful, but does not replace an explanation of your thought process (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request) (https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request))

caldermc-at-amazon commented:
Thank you, just updated the description. Please let me know if any additional information is required!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f90cf511610ebfd38ce2c52b04a3931af3083004
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ee225a46-795c-494b-966e-c12a8f9aa872%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented on packages/@aws-cdk/aws-apigateway/test/restapi.test.ts:
```diff
@@ -1096,4 +1096,21 @@ describe('restapi', () => {
     });
   });
+
+  test('"disableExecuteApiEndpoint" can disable the default execute-api endpoint for SpecRestApi', () => {
```
Let's also make sure that this properly sets this as false when it's false and that it's set as false when the property isn't set.

TheRealAmazonKendra reviewed (changes requested):
Thank you for your contribution. Just one small note below and we'll also need the conflicts resolved.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d9425f0b52e1313405f184b4828f393a5db5f877
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9e3aaa07-08ea-4c46-bf51-5b494a37b5fa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 09d1897587d3c83573b3d5aa2795dad26bc1dc01
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4a78871e-7def-4a61-815a-e213de7562d9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 27d46cb309497770e8588ee5ef22cfbfcd688f2f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=06afdc25-25e7-40b9-8c77-bced4c2b4285%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: e988c17acb57093f2f73747f5b80e5dfb7198b96
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=69881d69-01cc-4888-9b3c-b97444f8e894%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

caldermc-at-amazon commented:
Sorry for the delay, I have updated the diff to include true/false tests and resolve conflicts. Also it turns out that the CFN property should not be false (it will just be absent) when unspecified in the RestApi/SpecRestApi, so I removed the tests I originally added to check for that case.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 05a0aaf9dd8ecd88133ae41a778830459a918781
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=65d38f67-504a-4dee-b501-b2a17eb4f0cb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra reviewed (dismissed):
This looks great, however, I'm concerned about the fact that this change doesn't break any integration tests. This means that we're either lacking test coverage or that it's not passing this property correctly to a template. I suspect the former. To ease my mind, please add an integration test that uses this prop.

caldermc-at-amazon commented:
Sorry I'm not familiar with the process to do that. How would the integration test you're suggesting be any different than my tests in this pull request? To me the issue was a lack of explicit tests for this property in SpecRestApi, which I've now added. If you think this is a blocking concern please further elaborate on what you need changed. Thanks!

TheRealAmazonKendra commented:
Here is our documentation on [integration testing](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md). These tests will actually perform the deployment of this template and ensure that the deployment succeeds. This both tests that our templates are written correctly and ensures that changes don't have unintended consequences when they're in other packages that are used by this package. They are extremely important and I consider them mandatory when we have a change like this that isn't detected by an existing integration test.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9c0b2daa28728981c36cf48b2c782ae9e4e7b2c4
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=15522d84-d26a-49d6-b61c-38d00a16bca7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9c0b2daa28728981c36cf48b2c782ae9e4e7b2c4
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ceddb2d8-4dc1-4c4c-99e2-f5b9af6cc629%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9c0b2daa28728981c36cf48b2c782ae9e4e7b2c4
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=815697b8-c6d5-4237-a798-b56a6e3ce344%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 55e08ebe7108972274cc18494f420446efeddc35
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4b79589f-924a-44a7-95b7-8624548892fc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 76a59202e3156449f434ce78262b0a71e8168125
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d1a5f877-727a-4285-bea1-e1186c3aa6d4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 86d6dd449f6d56b21b1ba838c1f8f8c79c4f418f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c62a6b47-07f3-4e77-a4fc-9a95d3b7ad51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 39e6bb118932c5c110ae08248636a8e1243ce8fd
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cf90cf06-aa33-4d10-8d57-21e464ed6e2b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c90976443eedf327dd71c238ca59e9cf75ceaba9
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bd21cf61-8c97-4265-810d-7400024de19a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 116bf17cce864f40eacf4c41bd305e7428e23659
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0753209c-ad19-4d5b-827f-e4762b9116a0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: cfbd6e3854ec0e5acb301eb806f199144fd2cadc
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6eefec21-1941-4b88-a66b-7bb040c6c7d9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0756f26477aeb96ff228c044fcfad0e1202d9bc6
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4ebdba6b-6e53-4e3a-8e8c-8557dfbc115d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 2e9e2aff1264ed18162eade39e4f5cad672e02ae
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6700791e-e271-4381-a6c8-e792e2e5434f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ca5af482ca2909294873b97c0aaa819798f29c5e
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=34c8df19-de20-4f74-b9a5-15d009145912%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 4049d3aee0fa4fd09cfe6f2189cce84216198142
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=266b05b3-5bef-4292-9199-c8d0b07b0357%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

caldermc-at-amazon commented:
Hi @TheRealAmazonKendra, I've spent the last few days trying and failing to create/modify integration tests for the RestApi/SpecRestApi constructs. I'm sure with enough time I could figure it out, but even the existing integration tests are very minimalistic and don't cover true/false scenarios for most of the other fields. I appreciate insistence on high standards but in this case they are inconsistent because we recently allowed a [nearly-identical change](https://github.com/caldermc-at-amazon/aws-cdk/commit/03fc2bdbff5b3678e02b1017c575a681d5b9f786) through without integration test coverage. Please provide that same exemption here because the alternative is leaving this bug in CDK knowing there is an outstanding one-line PR that fixes it. An effort to improve the robustness of the RestApi/SpecRestApi integration tests sounds like a great idea, but is unnecessary scope-creep and not something I have the time to get done myself.

TheRealAmazonKendra commented:
> Hi @TheRealAmazonKendra, I've spent the last few days trying and failing to create/modify integration tests for the RestApi/SpecRestApi constructs. I'm sure with enough time I could figure it out, but even the existing integration tests are very minimalistic and don't cover true/false scenarios for most of the other fields. I appreciate insistence on high standards but in this case they are inconsistent because we recently allowed a [nearly-identical change](https://github.com/caldermc-at-amazon/aws-cdk/commit/03fc2bdbff5b3678e02b1017c575a681d5b9f786) through without integration test coverage. Please provide that same exemption here because the alternative is leaving this bug in CDK knowing there is an outstanding one-line PR that fixes it. An effort to improve the robustness of the RestApi/SpecRestApi integration tests sounds like a great idea, but is unnecessary scope-creep and not something I have the time to get done myself.

@caldermc-at-amazon Can you tell me more about what's causing the issue creating a test for this? I can help with troubleshooting/getting the test working. The fact that our tests are very minimalistic and inconsistent is precisely what concerns me.

Looking back at the previous exemption I provided, I think that I was in the wrong there. We have also been increasing our testing requirements on PRs recently so that one may have slipped in just before that change was made. Given that we only release once a week (usually on Thursdays but it can vary), we have some time to get tests in place before this would make it into a release anyway.

caldermc-at-amazon commented:
> > Hi @TheRealAmazonKendra, I've spent the last few days trying and failing to create/modify integration tests for the RestApi/SpecRestApi constructs. I'm sure with enough time I could figure it out, but even the existing integration tests are very minimalistic and don't cover true/false scenarios for most of the other fields. I appreciate insistence on high standards but in this case they are inconsistent because we recently allowed a [nearly-identical change](https://github.com/caldermc-at-amazon/aws-cdk/commit/03fc2bdbff5b3678e02b1017c575a681d5b9f786) through without integration test coverage. Please provide that same exemption here because the alternative is leaving this bug in CDK knowing there is an outstanding one-line PR that fixes it. An effort to improve the robustness of the RestApi/SpecRestApi integration tests sounds like a great idea, but is unnecessary scope-creep and not something I have the time to get done myself.
> 
> @caldermc-at-amazon Can you tell me more about what's causing the issue creating a test for this? I can help with troubleshooting/getting the test working. The fact that our tests are very minimalistic and inconsistent is precisely what concerns me.
> 
> Looking back at the previous exemption I provided, I think that I was in the wrong there. We have also been increasing our testing requirements on PRs recently so that one may have slipped in just before that change was made. Given that we only release once a week (usually on Thursdays but it can vary), we have some time to get tests in place before this would make it into a release anyway.

Sorry but again, I personally do not have additional cycles to spend on this. Gatekeeping one-line bug fixes is not a customer-obsessed mechanism to address technical debt. I agree that somebody should write a suite of thorough integration tests for this package, but that effort should be a decoupled "chore" task. In the context of this PR, CDK is strictly better with this change as-is than without it. If you'd prefer to leave this PR blocked on principle that's fine but someone else will need to continue it.

TheRealAmazonKendra commented:
It's been over a month so I'm going to go ahead and close this, with a few notes. As a reviewer to code changes, gate keeping is precisely my job description. While one-line changes may seem innocuous, we've all had that time where we made such a change and caused wide reaching customer impact. This is not a risk that we are choosing to take on behalf of our customers when we have a mechanism, i.e., integration tests, to guard against such a thing.

While I laud your emphasis on customer obsession, we too are customer obsessed in that we don't want to merge a change that could be a breaking change for our customers due to its untested state. In this case, we would need to show even more caution given that tests were attempted and the build failed each time.

I think we can agree that the CDK is, in your words "strictly better" waiting for verification of working code than merging code that previously showed failures when tests were written. This is not a matter of blocking on principal, but blocking on failed builds.
